src/syslog_mcp/tools/data_access/storage_queries.py

```
@@ -0,0 +1,218 @@
+    except Exception as e:
+        logger.error(f"Error loading saved searches: {e}")
+        return {"searches": {}, "created_at": datetime.utcnow().isoformat()}
+
+
+def save_searches_data(data: dict[str, Any]) -> bool:
+    """Save searches data to file."""
+    ensure_storage_dir()
+    
+    try:
+        data["updated_at"] = datetime.utcnow().isoformat()
+        with open(SAVED_SEARCHES_FILE, 'w') as f:
```

Ruff (0.12.2)

`open()` should be replaced by `Path.open()`

(PTH123)